src/include/duckdb/function/table/arrow/arrow_duck_schema.hpp

```diff
@@ -45,10 +51,23 @@ class ArrowType {
 	}
 	void ThrowIfInvalid() const;
 
-private:
+	static shared_ptr<ArrowType> GetTypeFromFormat(DBConfig &config, ArrowSchema &schema, string &format);
+
+	static shared_ptr<ArrowType> GetTypeFromSchema(DBConfig &config, ArrowSchema &schema);
+
+	static shared_ptr<ArrowType> CreateListType(DBConfig &config, ArrowSchema &child, ArrowVariableSizeType size_type,
+	                                            bool view);
+
+	static shared_ptr<ArrowType> GetArrowLogicalType(DBConfig &config, ArrowSchema &schema);
+
+	bool IsExtension() const;
+
+protected:
+	ArrowTypeEnum type_enum = BASE;
+
 	LogicalType type;
 	//! Hold the optional type if the array is a dictionary
-	unique_ptr<ArrowType> dictionary_type;
+	shared_ptr<ArrowType> dictionary_type;
```

Comment:
Maybe we can add a `shared_ptr<ArrowExtensionData>` to the `ArrowType` instead here, then we don't need to use inheritance or convert everything into a `shared_ptr`